### PR TITLE
#26: implementing JSON validation in AbstractBeanstalkMojo

### PR DESCRIPTION
**Pull request by prondzyn**
Issue #26. I updated the validation in AbstractBeanstalkMojo adding JSON files validation.

I also prepared puzzle in WarFile to consider that file removal.

### CODE REVIEW

**Comment by dmarkov** [comment]
@prondzyn Thanks, let me find someone who can review this pull request

**Comment by dmarkov** [comment]
@pinaf please review, thanks

File: src/main/java/com/jcabi/beanstalk/maven/plugin/WarFile.java [hunk 1]
```diff
@@ -51,6 +51,8 @@
  * @version $Id$
  * @since 1.0
  * @checkstyle DesignForExtensionCheck
+ * @todo #26 Check and remove this class as all the functionality is in
```
Comment (pinaf) [inline src/main/java/com/jcabi/beanstalk/maven/plugin/WarFile.java]: @prondzyn why not remove the methods for JSON validation already?

File: src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java [hunk 2]
```diff
@@ -80,19 +80,57 @@ public void executesSuccessfully() throws Exception {
...
     /**
+     * {@link AbstractBeanstalkMojo} can validate bad json.
+     * @throws Exception If something is wrong
+     */
+    @Test
+    public void validatesBadJson() throws Exception {
+        final DeployMojo mojo = new DeployMojo();
+        Assert.assertFalse(
+            mojo.validJson(new StringBuilder()
+                    .append("[\n")
```
Comment (pinaf) [inline src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java]: @prondzyn too much indentation here.

File: src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java [hunk 3]
```diff
@@ -80,19 +80,57 @@ public void executesSuccessfully() throws Exception {
...
+    /**
+     * {@link AbstractBeanstalkMojo} can validate good json.
+     * @throws Exception If something is wrong
+     */
+    @Test
+    public void validatesGoodJson() throws Exception {
+        final DeployMojo mojo = new DeployMojo();
+        Assert.assertTrue(
+            mojo.validJson(new StringBuilder()
+                    .append("{\n")
```
Comment (pinaf) [inline src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java]: @prondzyn too much indentation here.

File: src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java [hunk 4]
```diff
@@ -80,19 +80,57 @@ public void executesSuccessfully() throws Exception {
...
+        );
+    }
+
+    /**
+     * {@link AbstractBeanstalkMojo} can validate good json.
+     * @throws Exception If something is wrong
+     */
+    @Test
+    public void validatesGoodJson() throws Exception {
+        final DeployMojo mojo = new DeployMojo();
```
Comment (pinaf) [inline src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java]: @prondzyn we can inline this variable.

File: src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java [hunk 5]
```diff
@@ -80,19 +80,57 @@ public void executesSuccessfully() throws Exception {
     }
 
     /**
+     * {@link AbstractBeanstalkMojo} can validate bad json.
+     * @throws Exception If something is wrong
+     */
+    @Test
+    public void validatesBadJson() throws Exception {
+        final DeployMojo mojo = new DeployMojo();
```
Comment (pinaf) [inline src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java]: @prondzyn we can inline this variable.

File: src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java [hunk 6]
```diff
@@ -80,19 +80,57 @@ public void executesSuccessfully() throws Exception {
...
-        mojo.validYaml(new StringBuilder()
-            .append("Some illegal Prefix\n")
-            .append("Time: 2005-11-23 10:01:42 -5\n")
-            .append("Admin: ed\n")
-            .append("Messages:\n")
-            .append(" Hello is an error information\n")
-            .append(" for the configuration file\n").toString()
+        Assert.assertFalse(
+            mojo.validYaml(new StringBuilder()
+                    .append("Some illegal Prefix\n")
```
Comment (pinaf) [inline src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java]: @prondzyn too much indentation here.

File: src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java [hunk 7]
```diff
@@ -103,12 +141,15 @@ public void validatesBadYaml() throws Exception {
...
-        mojo.validYaml(new StringBuilder()
-            .append("Time: 2001-11-23 15:01:42 -5\n")
-            .append("User: ed\n")
-            .append("Warning:\n")
-            .append("  This is an error message\n")
-            .append("  for the log file\n").toString()
+        Assert.assertTrue(
+            mojo.validYaml(
+                new StringBuilder()
+                    .append("Time: 2001-11-23 15:01:42 -5\n")
```
Comment (pinaf) [inline src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java]: @prondzyn too much indentation here.

File: src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java [hunk 8]
```diff
@@ -80,19 +80,57 @@ public void executesSuccessfully() throws Exception {
...
+    }
+
+    /**
      * {@link AbstractBeanstalkMojo} can validate bad yaml.
      * @throws Exception If something is wrong
      */
-    @Test(expected = YAMLException.class)
+    @Test
     public void validatesBadYaml() throws Exception {
         final DeployMojo mojo = new DeployMojo();
```
Comment (pinaf) [inline src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java]: @prondzyn we can inline this variable.

File: src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java [hunk 9]
```diff
@@ -103,12 +141,15 @@ public void validatesBadYaml() throws Exception {
     @Test
     public void validatesGoodYaml() throws Exception {
         final DeployMojo mojo = new DeployMojo();
```
Comment (pinaf) [inline src/test/java/com/jcabi/beanstalk/maven/plugin/GenericMojoTest.java]: @prondzyn we can inline this variable.

**Comment by pinaf** [comment]
@prondzyn please see 9 comments above

**Comment by pinaf** [comment]
@prondzyn I don't understand why you didn't remove the JSON methods already and instead added that puzzle.

**Comment by prondzyn** [comment]
@pinaf I added the puzzle because the issue #26 was only about moving theJSON validation to AbstractBeanstalkMojo but I noticed that the entire WarFile is probably no longer needed. I thought it would be better to separately review the WarFile and it's usage to check if it can be removed but this exceeded time of #26 so I decided to create a puzzle. Correct me if I'm wrong - I'm still PDD newbie :)

**Comment by pinaf** [comment]
@prondzyn I don't know about removing the entire file. Let's focus on moving the methods for JSON and YAML validation - right now you just copied them.

**Comment by prondzyn** [comment]
@pinaf please take a look now

**Comment by pinaf** [comment]
@prondzyn I see your point now. Let's see what the architect thinks about removing `WarFile`. @yegor256 we need your architect opinion here - since the YAML and JSON validation methods have been moved, should we get rid of `WarFile` entirely?

**Comment by yegor256** [comment]
@prondzyn @pinaf looks like we don't need `WarFile` class any more... let's just remove it, I'm not against it at all

**Comment by pinaf** [comment]
@yegor256 thanks

**Comment by prondzyn** [comment]
@pinaf so the puzzle can remain?

**Comment by pinaf** [comment]
@prondzyn sure

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by pinaf** [comment]
@prondzyn looks like there is a problem with the puzzle format

**Comment by prondzyn** [comment]
@pinaf puzzle fixed

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge again

**Comment by dmarkov** [comment]
@pinaf thanks, I just added **33 mins** to your account, payment `AP-12V16145NM578114H`, 121 hours and 22 mins spent; review comments (c=18) added as a bonus; +33 to your rating, your total score is [+4215](http://www.netbout.com/b/35128?open=rating)

**Comment by dmarkov** [comment]
@rultor please deploy

**Pull request merged**